src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,256 @@
+                    Task<T> funcTask = funcToExecute(client.Client);
+                    tasks.Add(funcTask);
+                    tasks.Add(parallelAwait);
+
+                    var completedTask = await Task.WhenAny(tasks);
+                    if (!completedTask.Equals(parallelAwait))
+                    {
+                        if (completedTask.Status == TaskStatus.RanToCompletion)
+                        {
+                            client.BackoffEndTime = DateTimeOffset.UtcNow;
+                            client.FailedAttempts = 0;
+
+                            // Safe because the task is completed.
+                            return funcTask.Result;
+                        }
+                        else if (completedTask.Status == TaskStatus.Faulted)
+                        {
+                            tasks.Remove(completedTask);
+                            tasks.Remove(parallelAwait);
+
+                            if (IsRetryableException(completedTask.Exception))
+                            {
+                                lastException = completedTask.Exception;
+                                client.FailedAttempts++;
+                                TimeSpan backoffInterval = BackoffIntervalConstants.MinBackoffInterval.CalculateBackoffInterval(BackoffIntervalConstants.MaxBackoffInterval, client.FailedAttempts);
+                                client.BackoffEndTime = DateTimeOffset.UtcNow.Add(backoffInterval);
+                            }
+                            else
+                            {
+                                completedTask.Exception.Handle(e =>
+                                {
+                                    // propagate the original exception.
+                                    throw e;
+                                });
+                            }
+                        }
+                        else if (completedTask.Status == TaskStatus.Canceled)
+                        {
+                            // Throw expected OperationCanceledException
+                            completedTask.GetAwaiter().GetResult();
+                        }
+                    }
+                    else
+                    {
+                        tasks.Remove(parallelAwait);
+                    }
+                }
+            }
+
+            throw lastException;
+        }
+
+        private bool IsRetryableException(AggregateException ex)
+        {
+            int statusCode = 0;
+
+            IReadOnlyCollection<Exception> innerExceptions = ex.Flatten().InnerExceptions;
+
+            if (innerExceptions.Count > 0 && innerExceptions.All(ex => ex is RequestFailedException))
+            {
+                statusCode = (innerExceptions.Last() as RequestFailedException).Status;
+            }
+
+            switch (statusCode)
+            {
+                case (int)HttpStatusCode.RequestTimeout:       // 408
```

Comment:
I'm curious who sets 408 status code when a request timed out.

Reply:
We get a `RequestFailedException` from the SDK. SDK constructs this exception and sets the appropriate status.

Reply:
Wow, I'm so surprised. A network timeout will end up with 408. Have you tried it?

Reply:
I haven't yet, but let me try now.

Reply:
If there is a network error, the SDK will not make-up a 408 response code. 408 is a valid HTTP response code for timeout, but we don't utilize it. In the future we could, but we don't know. One example usage I've seen for this is clients that specify some suggested timeout in their request payload. The server can be smart enough to infer if the request can complete within their allotted time and respond with with 408 if not possible. 

We don't use it, most retry policies do include retries on it for future proofing.